index.js

```diff
@@ -71,25 +72,27 @@ AFRAME.registerComponent('terrain-model', {
     var terrainLoader = new THREE.TerrainLoader();
 
     // Create the plane geometry
-    var geometry = new THREE.PlaneGeometry(data.planeWidth, data.planeHeight, data.segmentsWidth, data.segmentsHeight);
+    var geometry = new THREE.PlaneBufferGeometry(data.planeWidth, data.planeHeight, data.segmentsWidth, data.segmentsHeight);
 
     // z-scaling factor
     var zPosition = data.zPosition;
 
     // The terrainLoader loads the DEM file and defines a function to be called when the file is successfully downloaded.
     terrainLoader.load(terrainURL, function (data) {
       // Adjust each vertex in the plane to correspond to the height value in the DEM file.
-      for (var i = 0, l = geometry.vertices.length; i < l; i++) {
-        geometry.vertices[i].z = data[i] / 65535 * zPosition;
+      var verts = geometry.attributes.position;
+      for (var vi = 2, di=0, l = verts.count*3; vi < l; vi+=3, di++) {
+        verts.array[vi] = data[di] / 65535 * zPosition;
       }
+      geometry.attributes.position.needsUpdate = true;
 
```

Comment:
This for loop is getting a bit hairy. Would you mind adding a comment so I'll be able to remember what these variables mean? Something like:

```js
/**
 * vi = The index of the current vertex's Z position in the plane geometry's position attribute array.
 * di = The index of the current data value.
 * l = The total number of vertices in the plane geometry.
 */
```